ios/RNShopify.m

```diff
@@ -0,0 +1,320 @@
+
+#import "RNShopify.h"
+#import "Buy.h"
+
+@implementation RNShopify {
+    RCTPromiseResolveBlock _resolve;
+    RCTPromiseRejectBlock _reject;
+}
+
+- (dispatch_queue_t)methodQueue
+{
+    return dispatch_get_main_queue();
+}
+
+RCT_EXPORT_MODULE()
+
+RCT_EXPORT_METHOD(initialize:(NSString *)domain key:(NSString *)key)
+{
+    //Application ID is always 8, as stated in official documentation from Shopify
+    self.client = [[BUYClient alloc] initWithShopDomain:domain
+                                                 apiKey:key
+                                                  appId:@"8"];
+}
+
+RCT_EXPORT_METHOD(getShop:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self.client getShop:^(BUYShop *shop, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code], error.localizedDescription, error);
+        }
+
+        resolve([shop JSONDictionary]);
+    }];
+}
+
+RCT_EXPORT_METHOD(getCollections:(NSUInteger)page resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self.client getCollectionsPage:page completion:^(NSArray<BUYCollection *> *collections, NSUInteger page, BOOL reachedEnd, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code], error.localizedDescription, error);
+        }
+
+        NSMutableArray *collectionDictionaries = [NSMutableArray array];
+
+        for (BUYCollection *collection in collections) {
+            [collectionDictionaries addObject: @{@"title":collection.title,@"id":collection.identifier}];
```

Comment:
Hehe, true :). I'm transforming the collection because there is a bug in the SDK's JSONDictionary method for collections. We use this method to get a dictionary for all other entity types, such as shop and products. Each Shopify object implements it. But, if collections have images, there is a circular reference bug. People reported this but the SDK team said that this method is primarily for their internal use. They're not planning to fix this bug because it doesn't bother them and the solution is not straightforward. 

Since we only need the _title_ and _id_ field, we just create the dictionary ourselves.